docs/BaremetalHost_ProvisioningState.dot

```diff
@@ -5,40 +5,30 @@ digraph BaremetalHost {
 
     Unmanaged [shape=doublecircle]
     Unmanaged -> Registering [label="BMC.* != \"\""]
-    Unmanaged -> Deleting1 [label="!DeletionTimestamp.IsZero()"]
-
-    Deleting1 [shape=point]
+    Unmanaged -> PoweringOffBeforeDelete [label="!DeletionTimestamp.IsZero()"]
 
     ExternallyProvisioned [label="Externally\nProvisioned"]
 
     Registering -> Inspecting [label="!externallyProvisioned && NeedsHardwareInspection()"]
     Registering -> Preparing [label="!externallyProvisioned && inspectionDisabled()"]
     Registering -> ExternallyProvisioned [label="externallyProvisioned"]
-    Registering -> Deleting2 [label="!DeletionTimestamp.IsZero()"]
-
-    Deleting2 [shape=point]
+    Registering -> PoweringOffBeforeDelete [label="!DeletionTimestamp.IsZero()"]
 
     ExternallyProvisioned -> Inspecting [label="!externallyProvisioned && NeedsHardwareInspection()"]
```

Comment:
I just discovered that if we change all these ones with && or || to e.g.
```suggestion
    ExternallyProvisioned -> Inspecting [label="!externallyProvisioned &&\nNeedsHardwareInspection()"]
```
it makes everything a bit more compact, and this one in particular fixes that diabolical intersection of the two edges at the top right.

Reply:
Like this?

![BaremetalHost_ProvisioningState](https://github.com/metal3-io/baremetal-operator/assets/206357/d8ba0af7-b8fd-45eb-8287-35f3a1f0a547)